src/common/crypto/hash.c

```diff
@@ -46,6 +51,7 @@ typedef struct CryptoHash
     MemContext *memContext;                                         // Context to store data
     const EVP_MD *hashType;                                         // Hash type (sha1, md5, etc.)
     EVP_MD_CTX *hashContext;                                        // Message hash context
+    MD5_CTX md5Context;                                             // MD5 context (used to bypass FIPS restrictions)
```

Comment:
Easier and safer to statically allocate, but it does use up a chunk of memory when not being used for MD5. Changed to dynamically allocate.

This also required using a designated initializer in MD5_Init() since this function does not do a complete initialization, i.e. it expects the passed struct to be pre-zeroed.

Reply:
OK. Can you please push the changes to this branch when you are done so I can do a final review? Thanks.